pilot/pkg/features/pilot.go

```diff
@@ -708,6 +708,8 @@ var (
 	PersistOldestWinsHeuristicForVirtualServiceHostMatching = env.Register("PERSIST_OLDEST_FIRST_HEURISTIC_FOR_VIRTUAL_SERVICE_HOST_MATCHING", false,
 		"If enabled, istiod will persist the oldest first heuristic for subtly conflicting traffic policy selection"+
 			"(such as with overlapping wildcard hosts)").Get()
+
```

Comment:
note the original issue, https://github.com/istio/istio/issues/46858, could be fixed without ECDS by making the config we send smaller

Reply:
No, that would still cause a drain event. It's not the same.

Reply:
> note the original issue, https://github.com/istio/istio/issues/46858, could be fixed without ECDS by making the config we send smaller

this will not fix #46858 , a post-processing still valuable.